plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -648,6 +661,30 @@ function MosaicMenuItem:update()
                 self.has_description = true
             end
 
+            if not self.menu.cover_info_cache then
+                self.menu.cover_info_cache = {}
+            end
+
+            local percent_finished, status
+            if bookinfo then
+                if self.menu.cover_info_cache[self.filepath] then
+                    percent_finished, status = unpack(self.menu.cover_info_cache[self.filepath])
+                else
+                    local docinfo = DocSettings:open(self.filepath)
+                    -- We can get nb of page in the new 'doc_pages' setting, or from the old 'stats.page'
+                    if docinfo.data.summary and docinfo.data.summary.status then
+                        status = docinfo.data.summary.status
+                    end
+                    percent_finished = docinfo.data.percent_finished
+                    self.menu.cover_info_cache[self.filepath] = {percent_finished, status}
```

Comment:
What is cached here is used for the current browsing session. But in a same browsing session, a user can change Display mode and switch from here to ListMenu item. So, you want to do and store exactely the same stuff that is done and store in ListMenuItem (even if you don't need it all).
(With storing only bits like in here, ListItem may crash as it won't get what it expects.)

Reply:
Done.